pkg/catalog/endpoint.go

```diff
@@ -38,15 +38,24 @@ func (mc *MeshCatalog) GetResolvableServiceEndpoints(svc service.MeshService) ([
 	return endpoints, nil
 }
 
-// ListEndpointsForServiceIdentity returns a list of endpoints that belongs to an upstream service accounts
-// from the given downstream identity's perspective
-// Note: ServiceIdentity must be in the format "name.namespace" [https://github.com/openservicemesh/osm/issues/3188]
-func (mc *MeshCatalog) ListEndpointsForServiceIdentity(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService) ([]endpoint.Endpoint, error) {
+// ListAllowedUpstreamEndpointsForService returns the list of endpoints over which the downstream client identity
+// is allowed access the upstream service
+func (mc *MeshCatalog) ListAllowedUpstreamEndpointsForService(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService) ([]endpoint.Endpoint, error) {
```

Comment:
In this issue https://github.com/openservicemesh/osm/issues/3190, we discussed about dropping the word `allowed` from method names. Basically we don't return endpoints that are not allowed.

Reply:
For permissive mode, everything is allowed. For SMI, a subset of endpoints is allowed. Hope that makes sense.